Lint the contrib/ directory

anoadragon453 opened the pull request:
This PR runs `isort`, `flake8` and `black` over the `contrib/` directory. We've decided not to include it in CI for now as we try to keep CI runtimes down, and this directory is rarely changed anyways.

Fixes https://github.com/matrix-org/synapse/issues/7910

clokep commented on contrib/experiments/test_messaging.py:
```diff
@@ -202,9 +199,9 @@ def on_receive_pdu(self, pdu):
             )
 
     # def on_state_change(self, pdu):
```
Looks like this code should just go away (and the second below it)?

anoadragon453 replied:
Good point.

anoadragon453 commented:
> I don't see a reason not to run this in CI `flake8 contrib` runs instantly for me.

Indeed, I was worried about CI times but you're right in the fact that the difference is negligible :)

anoadragon453 commented:
I've added `contrib` to the default list of directories linted by `lint.sh`. I've also added `synctl` as it was being checked by CI as well:

https://github.com/matrix-org/synapse/blob/1f98886212bc7a65c2c6370490bdbde25e5f3d7e/tox.ini#L129
